archethic-foundation/archethic-node
# Update node index on bootstrap in case of DB lost

# Description

There is some possibility that a running node shut down and lose it's DB (user intervention, hardware change ...). If it happen there is some case where the node cannot rejoin again the network, even with the same seed.

To avoid this, during the bootstrap, we fetch from the network the last index of the chain.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- Run 3 nodes
- Wait them to be authorized
- Stop a node and restart it after 1 minutes => It should create a new transaction in it's chain (2 tx)
- Stop again the same node
- Delete it's DB
- Restart it => It should create a new transaction in it's chain (3 tx)

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules
